bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/console/HueCommandExtension.java

```diff
@@ -78,41 +108,120 @@ public void execute(String[] args, Console console) {
+                                            continue;
+                                        }
+                                        String resourceId = resource.getId();
+                                        String idv1 = resource.getIdV1();
+                                        String thingLabel = resource.getName();
+                                        String comment = resource.getProductName();
+                                        String thingType = resourceType.name().toLowerCase();
+                                        String thingId = resourceId;
+
+                                        // special zone 'all lights'
+                                        if (resource.getType() == ResourceType.BRIDGE_HOME) {
+                                            thingLabel = clip2BridgeHandler
+                                                    .getLocalizedText(HueBindingConstants.ALL_LIGHTS_KEY);
+                                            comment = "Zone";
+                                            thingType = comment.toLowerCase();
+                                        }
+
+                                        Optional<Thing> legacyThingOptional = clip2BridgeHandler.getLegacyThing(idv1);
+                                        if (legacyThingOptional.isPresent()) {
+                                            Thing legacyThing = legacyThingOptional.get();
+                                            thingId = legacyThing.getUID().getId();
+                                            String legacyLabel = legacyThing.getLabel();
+                                            thingLabel = Objects.nonNull(legacyLabel) ? legacyLabel : thingLabel;
+                                        }
+
+                                        lines.put(thingLabel, String.format(FMT_THING, thingType, thingId, thingLabel,
+                                                resourceId, comment, idv1));
+                                    }
+                                    lines.entrySet().forEach(entry -> console.println(entry.getValue()));
+                                }
+                            }
+                            console.println("}");
+                            return;
+                    }
                 }
             }
-        } else {
-            printUsage(console);
         }
+        printUsage(console);
     }
 
     @Override
     public List<String> getUsages() {
         return Arrays.asList(new String[] { buildCommandUsage("<bridgeUID> " + USER_NAME, "show the user name"),
+                buildCommandUsage("<bridgeUID> " + APPLICATION_KEY, "show the API v2 application key"),
                 buildCommandUsage("<bridgeUID> " + SCENES, "list all the scenes with their id"),
+                buildCommandUsage("<bridgeUID> " + THINGS,
+                        "list all the API v2 device, room, and zone, things with their id"),
```

Comment:
One comma to suppress just before "things" or maybe rather "list all the API v2 device/room/zone things with their id" ?

Reply:
Hahah. As a French person, you would be delighted by the English style perfectionists debate about the 'Oxford Comma' (similar no doubt to some discussions within the Academie Francaise). But I will change it. Or something.. :)

Reply:
In case it is correct in English, do not change it. You know I am not very fluent.

Reply:
I changed it :)